### ISSUE TITLE
[bug]: Date Range Picker ： time from did not change after selecting a value

### ISSUE THREAD

**Issue by Gloria-wgs** [shadcn-ui/ui#5563]
### Describe the bug

Date Range Picker ： time from did not change after selecting a value

### Affected component/components

Date Range Picker

### How to reproduce

**Step 1** <img width="786" alt="image" src="https://github.com/user-attachments/assets/43472fa9-a027-4971-9b9d-2e7bfb98a126">
**Step 2**<img width="731" alt="image" src="https://github.com/user-attachments/assets/f7180af6-515d-4bd5-be4b-0e0041d7f60d">
**Step 3** <img width="717" alt="image" src="https://github.com/user-attachments/assets/d1e55664-efc2-4c22-95d5-ede7e106576a">

### System Info

```bash
browsers
```

### Before submitting

- [X] I've made research efforts and searched the documentation
- [X] I've searched for existing issues

**Comment by zhangyanling678** [comment]
yeah. If I select start time I'll never change it, but end time is correct.

**Comment by Shaker-Hamdi** [comment]
Any updates on this annoying bug?

**Comment by cnwangjie** [comment]
This is a bug of [react-day-picker](https://github.com/gpbl/react-day-picker).

They said it was fixed but it still exists. https://github.com/gpbl/react-day-picker/issues/2796

Is there a better component recommendation?

**Comment by rodgobbi** [comment]
@cnwangjie this is not a bug, it's the default behavior that can be customized.

But [react-day-picker](https://github.com/gpbl/react-day-picker) added the `resetOnSelect` prop in version [9.14.0](https://github.com/gpbl/react-day-picker/releases/tag/v9.14.0) that enables this behavior natively.
[Link to docs](https://daypicker.dev/selections/range-mode#reset-selection) with more info.
